design/one-pager-merge-jets-proposal.md

```diff
@@ -0,0 +1,370 @@
+# Merging AWS, Azure and GCP Providers
+
+* Owner: Muvaffak Onus (@muvaf)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Background
+
+[Terrajet] is a code generation framework that allows you to generate CRDs with
+a generic controller that is based on crossplane-runtime generic managed
+reconciler. The schema of the CRDs are derived from Terraform providers and the
+generic controller calls Terraform under the hood. With Terrajet, we're able to
+generate Crossplane providers that are on par with Terraform providers in terms
+of coverage in the matter of days. The API of the generated resources are fully
+compliant with [Crossplane Resource Model (XRM)][xrm-doc]. You can read more
+about its internals in [the deep dive blog post series][deep-dive-blogs].
+
+Before releasing the first Terrajet-based providers, we have had a discussion
+whether we'd like to have them as separate providers or have the
+Terrajet-generated CRDs added to the existing classic providers in the [provider
+strategy doc][strategy-doc]. The decision for initial releases of the first
+three Jet-based providers ([AWS][jet-aws], [Azure][jet-azure] and
+[GCP][jet-gcp]) was to have them as separate providers for the reasons stated
+[here](https://github.com/crossplane/crossplane/blob/master/design/design-doc-provider-strategy.md#decision-for-initial-releases).
+
+We called out that we'll revisit this decision after more usage and community
+feedback, which was lacking most probably because we were asking people (in
+Slack and in https://github.com/crossplane/crossplane/pull/2701) for their
+opinions about providers that they have not used yet. Today we have more data
+and more awareness in the community, and having two different implementations
+has been confusing for a lot of users, so a revisit of the decision regarding
+AWS, Azure and GCP is due.
+
+## Goals
+
+We have the following goals for _the big three_ providers:
+
+* The cloud vendors should maintain their Crossplane provider.
```

Comment:
A note for readers: those of us on the Crossplane Steering Committee are working on a charter for the project. We've talked about wanting vendors to own their own providers for a while but never really written it down. Expect it to be codified in the charter.

Reply:
So Crossplane Steering Committee or a provider generator will maintain the code base, and some mechanism will be responsible for patches from the new release of a Terraform Provider. Did I get the point?

Reply:
The goal is for vendor to maintain it in a way that works best for them, which could be under `crossplane` or `crossplane-contrib` or their own organization in Github.